robot/impl/resource_manager.go

```diff
@@ -920,17 +920,8 @@ func (manager *resourceManager) updateResources(
 	var allErrs error
 
 	// modules are not added into the resource tree as they belong to the module manager
-	for _, mod := range conf.Added.Modules {
-		// this is done in config validation but partial start rules require us to check again
-		if err := mod.Validate(""); err != nil {
-			manager.logger.CErrorw(ctx, "module config validation error; skipping", "module", mod.Name, "error", err)
-			continue
-		}
-		if err := manager.moduleManager.Add(ctx, mod); err != nil {
-			manager.logger.CErrorw(ctx, "error adding module", "module", mod.Name, "error", err)
-			continue
-		}
-	}
+	//nolint: errcheck
+	_ = manager.moduleManager.Add(ctx, conf.Added.Modules...)
```

Comment:
Won't this `_` swallow `errModularResourcesDisabled`? Furthermore, I think this should also be guarded by `if conf.Added.Modules != nil` or something. If we were in an untrusted environment, I think this would always return an error even if there were no added modules as we'd call `Add()` even if no modules were added.

Reply:
aren't we swallowing errors currently? I see we `continue` on `Validate` and `Add` errors.

oh interesting, what does it mean to be an untrusted environment in this context?